Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -1,15 +1,82 @@
-﻿namespace BootCamp.Chapter1
+﻿using System;
+namespace BootCamp.Chapter1
 {
     public static class ArrayOperations
     {
+        // *************private helpers**********
         /// <summary>
+        /// Test that Array is neither null or empty.
+        /// </summary>
+        /// <returns>True if array can be manipulated by standard functions.</returns>
+        /// 
+        private static bool ValidateArrayForManipulation(int[] array)
+        {
+            return ((array != null) && (array.Length != 0));
+        }
+
+        /// <summary>
+        /// Makes the array one index item shorter.
+        /// </summary>
+        /// <returns>A new array with (original length - 1)</returns>
+        /// 
+        private static int[] MakeAShorterArray(int[] array)
+        {
+            return new int[array.Length - 1];
+        }
+
+        /// <summary>
+        /// Makes a new array with one index with an int
+        /// </summary>
+        /// <returns>A new array with a single index with value {number}>
+        /// 
+        private static int[] AddOneNumberToNullArray(int number)
+        {
+            return new[] { number };
+        }
+
+         /// <summary>
+        /// Makes the array one index item longer.
+        /// </summary>
+        /// <returns>A new array with (original length+ 1)</returns>
+        private static int[] MakeALongerArray(int[] array)
+        {
+            return new int[array.Length + 1];
+        }
+
+        
+        /// <summary>
+        /// Confirm thar the index to be modified exists.
+        /// </summary>
+        /// <returns>True if index is valid
+        /// 
+        private static bool ConfirmThatTargetIndexToManipulateIsWithinTheRangeOfTheArray(int[] array, string action, int index)
+        {
+            return (action == "Remove") ? ((index < array.Length) && (index >= 0)) : ((index <= array.Length) && (index >= 0));
+        }
+
+        //public array operations
+
+         /// <summary>
         /// Sort the array in ascending order.
         /// If array empty or null- don't do anything.
         /// </summary>
         /// <param name="array">Input array in a random order.</param>
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            if (ValidateArrayForManipulation(array))
```

Comment:
If you had IsNullOrEmpty(array) here you could simply return within the if block. This leads to less scope levels and is generally preferable! :) The for loop would then be on the same level as the if keyword and therefor you did reduce complexity which is a good thing.